src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1981,8 +1984,14 @@ public async Task<List<SimpleVariableInformation>> GetParameterInfoOnly(AD7Threa
         //NOTE: eval is not called
         public async Task<List<ArgumentList>> GetParameterInfoOnly(AD7Thread thread, bool values, bool types, uint low, uint high)
         {
-            // If values are requested, request simple values, otherwise we'll use -var-create to get the type of argument it is.
-            var frames = await MICommandFactory.StackListArguments(values ? PrintValue.SimpleValues : PrintValue.NoValues, thread.Id, low, high);
+            // If SimpleValues excludes printing values for references to compound types,
+            // use SimpleValues (even if values are not requested) as it gets types too.
+            // Otherwise, if values are requested, use SimpleValues, but if not, the
+            // potential performance penalty of fetching values for references to compound
+            // types is too high, so use NoValues and follow up with -var-create to get
+            // the types.
+            PrintValue printValue = (_simpleValuesExcludesRefTypes || values) ? PrintValue.SimpleValues : PrintValue.NoValues;
```

Comment:
We only need `SimpleValues` if the caller is requested `values` or `types` so I think you want something like:

```
PrintValue printValue = values ? PrintValue.SimpleValues : PrintValue.NoValues;
if (!values && types)
{
    if (!_simpleValuesExcludesRefTypes.HasValue)
    {
        _simpleValuesExcludesRefTypes  = await this.MICommandFactory.SupportsSimpleValuesExcludesRefTypes();
    }
    if (_simpleValuesExcludesRefTypes.Value)
    {
        printValue = PrintValue.SimpleValues;
    }
}

Reply:
I pushed a fixup commit with your suggested changes.